### PR TITLE
[Feature] fix mtp mask for interleave size > 1 with pcp

### PR DESCRIPTION
**Pull request by SunnyLee151064**
### What this PR does / why we need it?
This pr support cp_kvcache_interleave_size > 1 + pcp + mtp. 
### Does this PR introduce _any_ user-facing change?
No
### How was this patch tested?

- vLLM version: v0.23.0
- vLLM main: https://github.com/vllm-project/vllm/commit/1f486d96a17303ce8db8e02be39545b2be338446

### CODE REVIEW

**Review by weiguihua2 — APPROVED** [review]

File: vllm_ascend/worker/pcp_utils.py [hunk 1]
```diff
@@ -1505,18 +1505,24 @@ def generate_mtp_attention_mask_for_decode(
...
 
-        max_indices = total_lens - 1
-        valid = max_indices >= cp_rank
+        # Interleave-aware per-rank KV length:
+        # base = L // I // W * I, remainder = L - base * W,
+        # local = base + clip(remainder - rank * I, 0, I)
+        base_k = total_lens // interleave_size // cp_size * interleave_size
+        remainder_k = total_lens - base_k * cp_size
+        k_lens = base_k + torch.clamp(remainder_k - cp_rank * interleave_size, 0, interleave_size)
+        valid = k_lens > 0
```
Comment (Tflowers-0129) [inline vllm_ascend/worker/pcp_utils.py:1521]: 补充大于1 的ut

**Review by realliujiaxu — APPROVED** [review]
LGTM
